litmus-portal/frontend/src/components/CustomStepper/index.tsx

```diff
@@ -95,10 +104,33 @@ function getStepContent(stepIndex: number): React.ReactNode {
 const CustomStepper = () => {
   const classes = useStyles();
   const [activeStep, setActiveStep] = React.useState(0);
+
+  const workflowData: WorkflowData = useSelector(
+    (state: RootState) => state.workflowData
+  );
+  const { yaml, weights } = workflowData;
+  const workflow = useActions(WorkflowActions);
+
   const steps = getSteps();
 
   const handleNext = () => {
     setActiveStep((prevActiveStep) => prevActiveStep + 1);
+    if (activeStep === 2) {
+      const tests = parsed(yaml);
+      const arr: experimentMap[] = [];
+      tests.forEach((test) => {
```

Comment:
We can use hash map here to reduce the time complexity from O(n2) to O(n)

Reply:
- make hashmap from weights
- declare empty experiment array
- loop over tests
- check if test exists in map.key
- if it exists in add name: test and weight: map.get(test)
- if it doesn't exist name: test and weight: 0
- change the global state entirely from experiments array from step 2